csrc/python_frontend/fusion_definition.cpp

```diff
@@ -228,6 +228,18 @@ std::string FusionDefinition::fusionIr() {
   return ss.str();
 }
 
+std::string FusionDefinition::userScheduleIr() {
+  NVF_CHECK(id().has_value(), "Invalid fusion definition!");
```

Comment:
IIUC, vvv
```
  class SomeFusion(FusionDefinition):
    def definition(self):
        # create some fusion
    def schedule(self):
        print(self._user_schedule_ir())

  fd = SomeFusion()
  fd.user_schedule_ir() # it would assert here with `Invalid fusion definition`?
```

one more nitpicking, this really isn't your problem, but we can probably use a better error message and probably a warning rather than an assert. I think all the other FusionDefinition methods defined in `nvfuser/__init__.py` suffers the same.

Keep it as-is in this PR. But I think we can use a clean up PR afterwards. Wdyt @kevinstephano